lib/diagnostic.js

```diff
@@ -5,22 +5,33 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  let answer = arguments[0];
+  for (let i = 1; i < arguments.length; i++) {
+    if (answer < arguments[i]) {
+      answer = arguments[i];
+    }
+  }
+  return(answer);
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-const sum = function sum() {
-
+const sum = function sum(x, y) {
+  return x + y;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let newArray = function(length, defaultValue) {
+    for (let i = 0; 1 < newArray.length; i++) {
+      newArray[i] = defaultValue;
+    }
+  };
+return newArray[i];
```

Comment:
Partially right - if you had actually made `newArray` a new array, and returned `newArray`, this would have worked.